### PR TITLE
Implement 'Manage My Instruments' Feature

### PR DESCRIPTION
**Pull request by moelgazzar**
The "Manage My Instruments" feature, allowing users to:
- View their listed instruments under `/my_instruments`.
- Edit and delete their instrument listings directly.
- Added a button to create a new instrument listing on the "My Instruments" page.

Screenshot 2024-10-13 at 17 54 26

### CODE REVIEW

File: app/controllers/instruments_controller.rb
Comment (p-guelfi) [inline app/controllers/instruments_controller.rb:1]: I don’t think we need the “authenticate_user!” here. We have a "before_action :authenticate_user!" applied globally on the ApplicationController. You only need to write a "skip_before_action" for the actions you want to have visible before authentication.

**Review by p-guelfi — CHANGES_REQUESTED** [review]
Good work! I would just change the thing about authentication in the instruments controller, and the link to "my instruments" from the navbar to the avatar dropdown.

File: app/controllers/instruments_controller.rb
Comment (moelgazzar) [reply app/controllers/instruments_controller.rb:1]: Good catch; I always get confused on this point; thanks for catching it. I just updated it!

Comment (p-guelfi) [reply app/controllers/instruments_controller.rb:1]: No worries!

**Review by p-guelfi — APPROVED** [review]

**Pull request merged**